deps/verifier/src/tpm/mod.rs

```diff
@@ -0,0 +1,439 @@
+        let mut pcrs = Vec::new();
+        for pcr_str in &self.pcrs {
+            if pcr_str.len() != 64 {
+                bail!(
+                    "PCR should be 64 hex characters (32 bytes), got {}",
+                    pcr_str.len()
+                );
+            }
+            let pcr_bytes = decode_hex(pcr_str)
+                .with_context(|| format!("Failed to decode PCR from hex: {}", pcr_str))?;
+            if pcr_bytes.len() != 32 {
+                bail!("PCR should be exactly 32 bytes, got {}", pcr_bytes.len());
+            }
+            let pcr_array: [u8; 32] = pcr_bytes.try_into()
+                .map_err(|_| anyhow!("PCR must be exactly 32 bytes"))?;
+            pcrs.push(pcr_array);
+        }
+        Ok(pcrs)
+    }
+}
+
+#[derive(Debug)]
+pub struct TpmVerifier {
+    trusted_ak_hashes: HashSet<Vec<u8>>,
+}
+
+impl Default for TpmVerifier {
+    fn default() -> Self {
+        let config = config::TpmVerifierConfig::default();
+        Self::new(config).unwrap_or_else(|_| Self {
+            trusted_ak_hashes: HashSet::new(),
+        })
+    }
+}
+
+impl TpmVerifier {
+    pub fn new(config: config::TpmVerifierConfig) -> Result<Self> {
+        let mut trusted_ak_hashes = HashSet::new();
+        if let Some(keys_dir) = config.trusted_ak_keys_dir {
+            let entries = fs::read_dir(&keys_dir)?.collect::<Result<Vec<_>, _>>()?;
+
+            debug!("TPM verifier trusted keys dir {:?}", keys_dir);
+
+            if entries.len() > config.max_trusted_ak_keys {
+                warn!(
+                    "Number of trusted AK keys ({}) exceeds the limit ({}). Only the first {} keys will be loaded.",
+                    entries.len(),
+                    config.max_trusted_ak_keys,
+                    config.max_trusted_ak_keys
+                );
+            }
+
+            for entry in entries.into_iter().take(config.max_trusted_ak_keys) {
+                let path = entry.path();
+                if path.is_file() {
+                    // Only process files with .pub extension
+                    if let Some(extension) = path.extension() {
+                        if extension != "pub" {
+                            continue;
+                        }
+                    } else {
+                        // Skip files without extensions
+                        continue;
+                    }
+
+                    // Try to read as text first to avoid UTF-8 errors on binary files
+                    let key_content = match fs::read_to_string(&path) {
+                        Ok(content) => content,
+                        Err(_) => {
+                            // Skip files that can't be read as UTF-8 text
+                            debug!("Skipping non-UTF8 file: {:?}", path);
+                            continue;
+                        }
+                    };
+
+                    let pkey = PKey::public_key_from_pem(key_content.as_bytes())
```

Comment:
we might need to change base64 to PEM [there](https://github.com/confidential-containers/trustee/pull/851/commits/5bda66d6de52544a22fe40b37ddfc8c93c3b1d67) ?